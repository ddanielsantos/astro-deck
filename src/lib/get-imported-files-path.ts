export function getImportedFilesPath(deck: string): string[] {
	const lines = deck.split("\n");

	const IMPORT_REGEX = /^\bimport\b/;

	const QUOTES_REGEX = /['"](.*)['"]/;

	return lines
		.filter((line) => IMPORT_REGEX.test(line))
		.map((imp) => {
			const match = imp.match(QUOTES_REGEX);

			if (match) {
				return match[1];
			}

			return "";
		});
}
