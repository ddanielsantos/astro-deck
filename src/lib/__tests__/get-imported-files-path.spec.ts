import { it, expect } from "vitest";

import { getImportedFilesPath } from "../get-imported-files-path.js";

const DECK_FILE = `import A from './a'
import B from './path/to/file.js'
import C from "./path/to/file2.ts"
import './path/to/file3.css'
importa

# import

## important
`;

it("should recognize JS imports", () => {
	const actual = getImportedFilesPath(DECK_FILE);

	const expected = [
		"./a",
		"./path/to/file.js",
		"./path/to/file2.ts",
		"./path/to/file3.css",
	];

	expect(actual).toEqual(expected);
	expect(actual).toMatchSnapshot();
});
