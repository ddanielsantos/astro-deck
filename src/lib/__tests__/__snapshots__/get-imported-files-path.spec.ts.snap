// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should recognize JS imports 1`] = `
[
  "./a",
  "./path/to/file.js",
  "./path/to/file2.ts",
  "./path/to/file3.css",
]
`;
